casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -145,7 +145,8 @@ object Genesis {
     case Array(ethAddress, initRevBalanceStr, _) =>
       Try(initRevBalanceStr.toLong) match {
         case Success(initRevBalance) =>
-          Right(Vault(RevAddress.fromEthAddress(ethAddress), initRevBalance))
+          val ethAddressWithoutPrefix = ethAddress.drop(2)
+          Right(Vault(RevAddress.fromEthAddress(ethAddressWithoutPrefix), initRevBalance))
```

Comment:
This should be done in AddressTools.fromEthAddress, which in turn should assert that the prefix is there, and is of correct form. I see you mean to strip `0x` to be stripped - I agree we could make it optional. In any case (prefix optional or mandatory) the prefix needs to be checked (so that we don't strip anything different than 0x).

Also, AddressTools should get corresponding tests for that.

Reply:
Stripping the `0x` prefix should probably even be the responsibility of `Base16` class, so that it works for all addresses we use. I won't insist on that though.